src/Data/Functor/Contravariant/Divise.hs

```diff
@@ -0,0 +1,244 @@
+
+#if MIN_VERSION_base(4,7,0)
+{-# LANGUAGE EmptyCase     #-}
+#endif
+
+-----------------------------------------------------------------------------
+-- |
+-- Copyright   :  (C) 2011-2015 Edward Kmett
+-- License     :  BSD-style (see the file LICENSE)
+--
+-- Maintainer  :  Edward Kmett <ekmett@gmail.com>
+-- Stability   :  provisional
+-- Portability :  portable
+--
+----------------------------------------------------------------------------
+module Data.Functor.Contravariant.Divise (
+    Divise(..)
+  , divised
+  , WrappedDivisible(..)
+  ) where
+
+import Control.Applicative
+import Control.Applicative.Backwards
+import Control.Arrow
+import Control.Monad.Trans.Error
+import Control.Monad.Trans.Except
+import Control.Monad.Trans.Identity
+import Control.Monad.Trans.List
+import Control.Monad.Trans.Maybe
+import qualified Control.Monad.Trans.RWS.Lazy as Lazy
+import qualified Control.Monad.Trans.RWS.Strict as Strict
+import Control.Monad.Trans.Reader
+import qualified Control.Monad.Trans.State.Lazy as Lazy
+import qualified Control.Monad.Trans.State.Strict as Strict
+import qualified Control.Monad.Trans.Writer.Lazy as Lazy
+import qualified Control.Monad.Trans.Writer.Strict as Strict
+
+import Data.Functor.Apply
+import Data.Functor.Compose
+import Data.Functor.Constant
+import Data.Functor.Contravariant
+import Data.Functor.Contravariant.Divisible
+import Data.Functor.Product
+import Data.Functor.Reverse
+
+#if MIN_VERSION_base(4,8,0)
+import Data.Monoid (Alt(..))
+#else
+import Data.Monoid (Monoid(..))
+#endif
+
+#if MIN_VERSION_base(4,9,0) && !MIN_VERSION_base(4,12,0)
+import Data.Semigroup (Semigroup(..))
+#endif
+
+#if MIN_VERSION_base(4,7,0) || defined(MIN_VERSION_tagged)
+import Data.Proxy
+#endif
+
+#ifdef MIN_VERSION_StateVar
+import Data.StateVar
+#endif
+
+#if __GLASGOW_HASKELL__ >= 702
+#define GHC_GENERICS
+import GHC.Generics
+#endif
+
+-- | The contravariant analogue of 'Apply'; it is
+-- 'Divisible' without 'conquer'.
+--
+-- If one thinks of @f a@ as a consumer of @a@s, then 'divise' allows one
+-- to handle the consumption of a value by splitting it between two
+-- consumers that consume separate parts of @a@.
+--
+-- 'divise' takes the "splitting" method and the two sub-consumers, and
+-- returns the wrapped/combined consumer.
+--
+-- All instances of 'Divisible' should be instances of 'Divise' with
+-- @'divise' = 'divide'@.
+--
+-- The guarantee that a function polymorphic over of @'Divise' f@ provides
+-- that @'Divisible' f@ doesn't that any input consumed will be passed to at
+-- least one sub-consumer; it won't potentially disappear into the void, as
+-- is possible if 'conquer' is available.
+--
+-- Mathematically, a functor being an instance of 'Divise' means that it is
+-- "semgroupoidal" with respect to the contravariant (tupling) Day
+-- convolution.  That is, it is possible to define a function @(f `Day` f)
+-- a -> f a@ in a way that is associative.
+class Contravariant f => Divise f where
```

Comment:
I always expect this to be called `Divide`, but I suppose then if you try to follow that convention into the actual function names, you hit clashes with the classes in `contravariant`.

Reply:
indeed, that was my main reasoning as well :'(  overall the names are a bit ad-hoc; #90 would have fixed things up in this regard but it seems it has been dropped.

Reply:
Maybe someone else could pick it up?